packages/web/src/components/edit-track/EditTrackForm.tsx

```diff
@@ -327,6 +327,11 @@ const TrackEditForm = (
   const { onOpen: openWaitforDownload } = useWaitForDownloadModal()
 
   const onClickDownload = useCallback(() => {
+    if (!initialTrackValues.track_id) {
+      console.error('Cannot download track without track ID')
```

Comment:
I chose console errors here because this should be impossible unless you make a coding mistake. This handler can't be called unless you're editing an existing track.